src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -867,63 +867,87 @@ private int ReadBuffer(Span<char> userBuffer, out bool readToUserBuffer)
 
         private async Task<string?> ReadLineAsyncInternal()
         {
+            static char[] ResizeOrPoolNewArray(char[]? array, int atLeastSpace)
+            {
+                if (array == null)
+                {
+                    return ArrayPool<char>.Shared.Rent(atLeastSpace);
+                }
+
+                char[] newArr = ArrayPool<char>.Shared.Rent(array.Length + atLeastSpace);
+                Array.Copy(array, newArr, array.Length);
+                ArrayPool<char>.Shared.Return(array);
```

Comment:
This returns rentedArray.  If between here and to = newArr later an exception occurs, it'll end up double-returning the array.

Reply:
Have a look at it again